Vim plugins

jhol opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

jhol committed:
vimPlugins.vim-cmake: init at 2023-06-27

jhol committed:
vimPlugins.SmartCase: init at 2023-06-27

jhol committed:
vimPlugins.error-marker: init at 2023-06-27

jhol committed:
vimPlugins.vim-autosource: init at 2023-06-27

jhol committed:
vimPlugins.vim-gas: init at 2023-06-27

jhol committed:
vimPlugins.tinykeymap: init at 2023-06-27

figsoda commented:
just a quick note that this might run into merge conflicts with #240038 

also you have 2 `vimPlugins: update` commit, would be nice to rebase them into 1 commit

jhol commented:
> just a quick note that this might run into merge conflicts with #240038
> 
> also you have 2 `vimPlugins: update` commit, would be nice to rebase them into 1 commit

Happy to rebase/regenerate on top of https://github.com/NixOS/nixpkgs/pull/240038. The reason for having two update commits was to separate background updates from the updates associated with my commits. Though I'm happy to redo it with a single update if this is more conventional.

figsoda commented:
I've gone ahead and merged #240038

> The reason for having two update commits was to separate background updates from the updates associated with my commits

Sounds good to me

jhol committed:
vimPlugins: update

jhol committed:
vimPlugins.nvim-treesitter: update grammars

jhol commented:
> I've gone ahead and merged #240038
> 
> > The reason for having two update commits was to separate background updates from the updates associated with my commits
> 
> Sounds good to me

Now updated, and rebased with just one update operation.
